common/intensityscaling.cpp

```diff
@@ -7,21 +8,22 @@ using namespace std;
 #include "binaryio.hpp"
 
 
-void read_image_IntensityScaling (string inputFilename, string outputFilename, int newIntensity){
-    ifstream inputImageFile(inputFilename);
+void read_image_intensity_scaling (int newIntensity){
+    ifstream inputImageFile(getInFile());
     if(!inputImageFile.is_open()) {
         cerr << "Failed to open input file\n";
         exit(-1);
     }
-    ofstream outputImageFile(outputFilename);
+    ofstream outputImageFile(getOutFile());
     if(!outputImageFile.is_open()) {
         cerr << "Failed to open output file\n";
         exit(-1);
     }
     string magic_word;
     int width, height, intensity;
     inputImageFile >> magic_word >> width >> height >> intensity;

```

Comment:
Why not use get_image_metadata here? Would it not be the same speed + cleaner?

Reply:
I agree. @100496948 can you make these changes and make a pull request

Reply:
Yeah, I'll work on it. I'm not sure if i'll be able to do it today